src/ngSanitize/sanitize.js

```diff
@@ -518,8 +524,17 @@ function $SanitizeProvider() {
         stripCustomNsAttrs(nextNode);
       }
 
-      node = node.nextSibling;
+      node = getNonDescendant('nextSibling', node);
+    }
+  }
+
+  function getNonDescendant(propName, node) {
+    // An element is clobbered if its `propName` property points to one of its descendants
+    var nextNode = node[propName];
+    if (nextNode && nodeContains.call(node, nextNode)) {
```

Comment:
I don't know of another usecase that would be affected other than `<form>` elements. (`document` is affected too, but we can't access `document` inside the htmlParsers afaict.)
I can't be 100% sure though. DOM has many dark corners :smiley: